### PR TITLE
Update python3-aiohttp_socks to 0.5.5 and add dependencies

### PR DESCRIPTION
**Pull request by ar-jan**
#### General
- [x ] This ~~is~~ includes ~~a~~ new package**s** and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Overview
- Electrum 4 depends on a newer version of `aiohttp_socks` than is currently packaged ([electrum-4.0.9 update here](https://github.com/void-linux/void-packages/pull/28305)).
- `aiohttp_socks` >= 0.5.4 depends on `python-socks`, added as new package here.
- `python-socks` has optional dependencies on `python-curio` and `python-trio`; however these libraries are needed in order to pass `python-socks` tests, so they are also added as new packages here.
- `python-trio` depends on `sniffio` and `outcome` libraries, also added as new packages here.
- `python-trio` tests also depend on `pyOpenSSL`, `trustme`, and `astor` which are not available, so its tests are skipped for now.

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (ARCH-LIBC)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

Dependencies added:

    python3-socks-1.2.0
    python3-curio-1.4
    python3-trio-0.18.0
    python3-outcome-1.1.0
    python3-sniffio-1.2.0

### CODE REVIEW

File: srcpkgs/python3-python-socks/template [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'python3-python-socks'
+pkgname=python3-python-socks
```
Comment (sgn) [inline srcpkgs/python3-python-socks/template:2]: I prefer to name it `python3-socks`. However, I prefer to include this package when it's actually need it.

**Comment by ar-jan** [comment]
@sgn python-socks is needed for python3-aiohttp_socks >= 0.5.4. Electrum 4.x requires aiohttp_socks >= 0.3 but currently 0.2.2 is available in void-packages. So upgrading to latest aiohttp_socks seemed best.

Or should I try to combine all of these in one pull request? (adding python-socks and dependencies, upgrading aiohttp_socks, and upgrading electrum). Doing this in three steps seemed simpler.

**Comment by ar-jan** [comment]
I've renamed to python3-socks, and included python3-aiohttp_socks update in this pull request.

Update for Electrum is here #28305 because there's a question about which version of secp256k1 to use as dependency.

File: srcpkgs/python3-aiohttp_socks/template [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-version=0.2.2
-revision=3
+version=0.5.5
+revision=1
 wrksrc="aiohttp_socks-${version}"
 build_style=python3-module
-pycompile_module="aiohttp_socks"
 hostmakedepends="python3-setuptools"
-depends="python3-aiohttp"
+depends="python3-aiohttp python3-socks>=1.0.1"
```
Comment (Chocimier) [inline srcpkgs/python3-aiohttp_socks/template:8]: There was never unmatching version in tree, please remove version spec.

File: srcpkgs/python3-curio/template [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+hostmakedepends="python3-setuptools"
+short_desc="Coroutine-based library for concurrent programming using async/await"
+maintainer="Arjan Mossel <arjanmossel@gmail.com>"
+license="BSD-3-Clause"
+homepage="https://github.com/dabeaz/curio"
+distfiles="${PYPI_SITE}/c/curio/curio-${version}.tar.gz"
+checksum=57edce81c837f3c2cf42fbb346dee26e537d1659e6605269fb13bd179e068744
+
+do_check() {
+	:
```
Comment (Chocimier) [inline srcpkgs/python3-curio/template:16]: These packages pass tests with python3-pytest in checkdepends and do_check defined as `PYTHONPATH=$(cd build/lib* && pwd) pytest`, please update.

**Review by Chocimier — COMMENTED** [review]
Every python package should depend on either other module or interpreter.
Some packages depend on other modules, eg. trio on attrs.

**Comment by ar-jan** [comment]
I've updated the the status and explanation about the dependencies to the issues:

- Electrum 4 depends on a newer version of `aiohttp_socks` than is currently packaged ([electrum-4.0.9 update here](https://github.com/void-linux/void-packages/pull/28305)).
- `aiohttp_socks` >= 0.5.4 depends on `python-socks`, added as new package here.
- `python-socks` has optional dependencies on `python-curio` and `python-trio`; however these libraries are needed in order to pass `python-socks` tests, so they are also added as new packages here.
- `python-trio` depends on `sniffio` and `outcome` libraries, also added as new packages here.
- `python-trio` tests also depend on `pyOpenSSL`, `trustme`, and `astor` which are not available, so its tests are skipped for now.

**Comment by ar-jan** [comment]
I can build all included new packages and the updated package locally with `xbps-src pkg -Q`, but for some reason the CI keeps failing on some arches. Any clues?

**Comment by ar-jan** [comment]
Closing, updated #28305 with these dependencies.

**Pull request closed without merging**